engine/extension/src/dmsdk/extension/extension.h

```diff
@@ -247,6 +247,38 @@ namespace dmExtension
      */
     void UnregisterAndroidOnActivityResultListener(OnActivityResult listener);
 
+    /*# OnCreateListener callback typedef
+     *
+     * onCreate callback function type.
+     * Used with RegisterAndroidOnCreateListener() and UnregisterAndroidOnCreateListener()
+     *
+     * @typedef
+     * @name OnCreateListener
+     * @param env [type:void*]
+     * @param activity [type:void*]
+     */
+    typedef void (*OnCreateListener)(void* env, void* activity);
```

Comment:
On second thought, I think the name is a bit too generic.
Perhaps `OnActivityCreate(JNIEnv*, ANativeActivity*)` ?

Reply:
To make it build for other platforms:
```
#if defined(ANDROID)
struct JNIEnv;
struct ANativeActivity;
#endif
```

os, like in android.h
```
#if defined(ANDROID)
#include <jni.h>
#include <android_native_app_glue.h> // For ANativeActivity
#endif
```

Reply:
Now that I mentioned it, I wonder if this function is actually better suited for `android.h`. It's a new file, and we probably want/need similar files for other platforms later on. It allows us to make more specialized functions, without.

Are we really going to need this generic callback on other platforms? If so, we can take it as it comes.
But for now, I'd rather move it into the Android space.

E.g. also rename functions `glfwAndroid*`

What do you think?

Reply:
@JCash is it fine to have 
```
#if defined(ANDROID)
#include <jni.h>
#include <android_native_app_glue.h> // For ANativeActivity
#endif
```

in `glfw.h ` ?

Reply:
I think it's done!
Thank you!!!